cylc/flow/config.py

```diff
@@ -2672,3 +2726,24 @@ def upgrade_clock_triggers(self):
             taskdef = self.get_taskdef(task_name)
             for seq in taskdef.sequences:
                 taskdef.add_xtrig_label(label, seq)
+
+    def _upg_wflow_event_names(self) -> None:
+        """Upgrade any Cylc 7 workflow handler/mail events names."""
+        for setting in ('handler events', 'mail events'):
+            event_names: Optional[List[str]] = self.cfg['scheduler']['events'][
+                setting
+            ]
+            if not event_names:
+                continue
+            upgraded: Dict[str, str] = {}
+            for i, event in enumerate(event_names):
+                if event in WorkflowEventHandler.EVENTS_DEPRECATED:
+                    event_names[i] = upgraded[event] = (
+                        WorkflowEventHandler.EVENTS_DEPRECATED[event]
+                    )
+            if upgraded and not cylc.flow.flags.cylc7_back_compat:
+                LOG.warning(
+                    f"{upgrader.DEPR_MSG}\n"
+                    f" * (8.0.0) [scheduler][events][{setting}] "
+                    + ', '.join(f'{k} -> {v}' for k, v in upgraded.items())
+                )
```

Comment:
[minor] We could potentially make `depr_options` a dict of `{deprecated_opt: replacement}` and handle this in the parsec layer if we thought this functionality would be reused elsewhere.